src/frontend/src/flows/registerDisabled.ts

```diff
@@ -1,47 +1,55 @@
-    <h1>Create a new Internet Identity Anchor</h1>
-    ${warnBox({
-      title: "Registration Disabled",
-      message: html`<p>
-          You are <b>not</b> browsing this website on the expected URL:
-          <a href="https://identity.ic0.app">https://identity.ic0.app</a>. For
-          security reasons creation of new Internet Identity anchors is disabled
-          on this origin.
-        </p>
-        <p>
-          Please switch to
-          <a href="https://identity.ic0.app">https://identity.ic0.app</a> to
-          register a new Internet Identity anchor.
-        </p>
-        <p>
-          If you were redirected here by another website, please inform the
-          developers. More information is provided
-          <a
-            href="https://forum.dfinity.org/t/internet-identity-proposal-to-deprecate-account-creation-on-all-origins-other-than-https-identity-ic0-app/9760"
-            >here</a
-          >.
-        </p>`,
-    })}
-    <button id="deviceAliasCancel">Cancel</button>
+    <hgroup>
+      <h1 class="t-title t-title--main">
+        Create a new Internet Identity Anchor
+      </h1>
+    </hgroup>
+    <div class="l-stack">
+      ${warnBox({
+        title: "Registration Disabled",
+        message: html`<p class="t-paragraph t-lead">
+            You are <b class="t-strong">not</b> browsing this website on the
+            expected URL:
+            <a class="t-link" href="https://identity.ic0.app">https://identity.ic0.app</a>. For
+            security reasons creation of new Internet Identity anchors is
+            disabled on this origin.
+          </p>
+          <p class="t-paragraph">
+            Please switch to
+            <a class="t-link" href="https://identity.ic0.app">https://identity.ic0.app</a> to
+            register a new Internet Identity anchor.
+          </p>
+          <p class="t-paragraph">
+            If you were redirected here by another website, please inform the
+            developers. More information is provided
+            <a
+              class="t-link"
+              href="https://forum.dfinity.org/t/internet-identity-proposal-to-deprecate-account-creation-on-all-origins-other-than-https-identity-ic0-app/9760"
+              >here</a
+            >.
+          </p>`,
+      })}
+    </div>
+    <div class="l-stack">
+      <button class="c-button" @click="${onCancel}" id="deviceAliasCancel">
+        Cancel
+      </button>
+    </div>
   </div>
 `;
 
-export const registerDisabled = async (): Promise<null> => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent, container);
-  return init();
-};
-
-const init = (): Promise<null> =>
-  new Promise((resolve) => {
-    const deviceAliasCancel = document.getElementById(
-      "deviceAliasCancel"
-    ) as HTMLButtonElement;
-    deviceAliasCancel.onclick = () => {
-      resolve(null);
-    };
+export const registerDisabled = (): Promise<null> => {
```

Comment:
`Promise<null>` is correct, you do something with the `null` result or it should actually be `Promise<void>`?

Reply:
It's actually expecting a `null`, which means "user canceled the flow". Agree it's a bit ugly that it trickles down here and the caller should probably translate `null` into `void` but that'd involve some bigger changes.

Reply:
If you would review the PR, you would probably suggest to add a comment then 😉. 
Thanks for the feedback, gotcha